casepro/pods/base.py

```diff
@@ -0,0 +1,31 @@
+from confmodel import fields, Config as ConfmodelConfig
+from django.apps import AppConfig
+
+
+class PodPlugin(AppConfig):
+    pass
+
+
+class PodConfig(ConfmodelConfig):
+    '''
+    This is the config that all pods should use as the base for their own
+    config.
+    '''
+    index = fields.ConfigInt(
+        "A unique identifier for the specific instance of this pod."
+        "Automatically determined and set in the pod registry.",
+        required=True)
+
+
+class Pod(object):
+    '''
+    The base class for all pod plugins.
+    '''
+    config_cls = PodConfig
+    url_patterns = []
+    name = 'Default pod name'
+    controller = None
+    directive = None
```

Comment:
We'll need to add configuration for paths to scripts and stylesheets here, but probably makes more sense to do that in a separate PR (happy to take that on once this PR lands).